### PR TITLE
MOB-67 Add join event endpoint

### PR DESCRIPTION
**Pull request by OmarVidalPerezAccenture**
MOB-67 Add join event endpoint.
Create EmptyBodyObject for empty responses in NetworkRequest Added method updateEvent(id: String, isUserJoined: Bool) in LocalEventsDataSource to update stored events. Added method joinEvent(id: String) in RemoteMyEventsDataSource to join remote event Create unit tests

### CODE REVIEW

File: EsmorgaiOS/App/Data/DataSource/Events/LocalEventsDataSource.swift [hunk 1]
```diff
@@ -44,6 +45,22 @@ class LocalEventsDataSource: LocalEventsDataSourceProtocol {
         return ()
     }
 
+    func updateEvent(id: String, isUserJoined: Bool) async throws {
```
Comment (andreslopezsarmiento) [inline EsmorgaiOS/App/Data/DataSource/Events/LocalEventsDataSource.swift:48]: shouldn't `updateEvent` receive an event as parameter? Having a generic name for a specific functionality may be confusing.

File: EsmorgaiOSTests/App/Data/DataSource/Events/RemoteMyEventsDataSourceTests.swift [hunk 2]
```diff
@@ -68,6 +68,32 @@ final class RemoteMyEventsDataSourceTests {
...
+        stubJoinEventRequest(file: "mock_empty_file.json")
+        do {
+            _ = try await sut.joinEvent(id: "123")
+        } catch {
+            Issue.record("Expected error not be thrown")
+        }
+    }
+
+    @Test
+    func test_given_2() async {
```
Comment (andreslopezsarmiento) [inline EsmorgaiOSTests/App/Data/DataSource/Events/RemoteMyEventsDataSourceTests.swift:84]: Use a more descriptive name

Comment (OmarVidalPerezAccenture) [reply EsmorgaiOSTests/App/Data/DataSource/Events/RemoteMyEventsDataSourceTests.swift:84]: yep, I forgot to set the name haha

File: EsmorgaiOS/App/Data/DataSource/Events/LocalEventsDataSource.swift (hunk 1 above)
Comment (OmarVidalPerezAccenture) [reply EsmorgaiOS/App/Data/DataSource/Events/LocalEventsDataSource.swift]: it was to not be redundant with the parameter of the method, but yep, can be confusing, I change the name to `updateIsUserJoinedEvent`

**Pull request merged**